vassal-app/src/main/java/VASSAL/build/module/GameState.java

```diff
@@ -1048,6 +1126,9 @@ public void loadGameInForeground(final String shortName,
     final GameModule g = GameModule.getGameModule();
     g.warn(Resources.getString("GameState.loading", shortName));  //$NON-NLS-1$
 
+    //TODO It would be nice to unblock EDT long enough to have the above loading message get displayed.
+    //TODO I guess it would mean right here we'd need to launch some "do in 10 milliseconds" thing, but then make sure that the below was then executed on EDT?
```

Comment:
The problem here is that you're _already_ on the EDT here, so if you post a Runnable to the EDT queue, it won't run until after this method finishes.

What you'd need to do is post your Runnable to display the message, then put the rest of this method into another Runnable that gets posted----but that's assuming that the caller of this method is not waiting on the completion of everything in it, which is an assumption you'd need to verify first.

Reply:
The trick here is the whole idea of "loadGameInForeground" is that we *want* to block all the other parts of Vassal from doing anything while we're loading a game over top of the current game (because there were all manner of horrific bugs we fixed this way). But ideally I'd like to be able to tell Swing to "take 10 milliseconds and just put this one last message up first, so the user knows we're attempting the load". I just haven't figured out how *from inside the EDT* to tell it to take a 10 ms break but _then block the EDT while the file loads_. Can you think of any way forward on that?